### PR TITLE
Update download mode detection

### PR DESCRIPTION
**Pull request by DaneSlattery**
Previously, this failed to detect download mode during this regex match in the following message.

```
ESP-ROM:esp32s3-20210327
    Build:Mar 27 2021
    rst:0x1 (POWERON),boot:0x0 (DOWNLOAD(USB/UART0))
    waiting for download

```

`.*` only matches on any character EXCEPT line breaks,  where as `[\s\S]` will match on anything.

### CODE REVIEW

**Review by SergioGasquez — COMMENTED** [review]
Thanks for your contribution! Although its only one line, it probably deserves a changelog entry. Mind adding it?

File: espflash/src/connection/mod.rs [hunk 1]
```diff
@@ -173,7 +173,8 @@ impl Connection {
 
             let read_slice = String::from_utf8_lossy(&buff[..read_bytes as usize]).into_owned();
 
-            let pattern = Regex::new(r"boot:(0x[0-9a-fA-F]+)(.*waiting for download)?").unwrap();
+            let pattern = 
+                Regex::new(r"boot:(0x[0-9a-fA-F]+)([\s\S]*waiting for download)?").unwrap();
```
Comment (SergioGasquez) [inline espflash/src/connection/mod.rs:177]:
```suggestion
            let pattern =
                 Regex::new(r"boot:(0x[0-9a-fA-F]+)([\s\S]*waiting for download)?").unwrap();
```

`rustfmt` is still not happy

Comment (DaneSlattery) [reply espflash/src/connection/mod.rs]: thank you

**Comment by jessebraham** [comment]
What is going on with CI? It was passing before, I just updated the Changelog. Sorry about that 😅

**Comment by SergioGasquez** [comment]
> It was passing before, I just updated the Changelog. 

I think it wasn't passing before, `rustfmt` is complaining about some spaces :(

**Comment by DaneSlattery** [comment]
ci is green!

**Review by NeaguSorin — APPROVED** [review]

**Review by NeaguSorin — APPROVED** [review]

**Review by SergioGasquez — APPROVED** [review]
Thanks for fixing it!

**Pull request merged**